react/src/hooks/groups/useGroupsCreateMutation.tsx

```diff
@@ -0,0 +1,28 @@
+import { useMutation, useQueryClient } from "react-query";
+import { Group } from "../../typings";
+import { addToCachedList, changeFetch } from "../utils";
+
+async function postNewGroup(newGroup: Group) {
+    return changeFetch("/api/groups", "POST", {
+        ...newGroup,
+        group_code: newGroup.group_code.toLowerCase(),
+    });
+}
+
+/**
+ * Return a mutation object for POST /api/groups.
+ *
+ * Used for creating a new group.
+ */
+export function useGroupsCreateMutation() {
```

Comment:
This only creates one group so I'm wondering if the naming convention makes sense if we're naming mutations after what they do instead of the HTTP verb.

Reply:
I saw it as "groups" is the API name, and "create" is the general action. I can change it to be more specific.